sdk/v2/core/projects.go

```diff
@@ -340,3 +341,15 @@ func (p *projectsClient) Authz() AuthzClient {
 func (p *projectsClient) Secrets() SecretsClient {
 	return p.secretsClient
 }
+
+// ValidateProjectID checks if a given id is valid.
+func ValidateProjectID(id string) error {
+	regex := regexp.MustCompile(`^[a-z][a-z\d-]*[a-z\d]$`)
+	idMatch := regex.MatchString(id)
+	if !idMatch || len(id) < 3 || len(id) > 63 {
+		return fmt.Errorf("invalid value %q for --id flag"+
+			" (3-63 characters, first char must be"+
+			" a letter, lowercase letters only)", id)
```

Comment:
Since this has now moved out of the CLI and into the SDK, it could be used in any number of ways that have nothing to do with the CLI, so this error message probably needs to be more generic now-- i.e. no mention of a specific flag.